packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -29,4 +29,37 @@ export class GOVUKFrontendComponent {
       throw new SupportError()
     }
   }
+
+  /**
+   *
+   * @protected
+   * @param {Schema} schema
+   * @param {{ [key: string]: number | string | object | undefined}} config
```

Comment:
TypeScript won't help here as you'll see that `config` has become type `any`

Problem comes from adding the type `object` which is always awful in JavaScript 😭

```
typeof [] === 'object'
typeof null === 'object'
typeof new Map() === 'object'
```

Thankfully we have a "component configs" type already, just needs moving

I’d suggest we [move `ConfigKey`](https://github.com/alphagov/govuk-frontend/blob/main/shared/helpers/puppeteer.js#L257) into [**packages/govuk-frontend/src/govuk/all.mjs**](https://github.com/alphagov/govuk-frontend/blob/main/packages/govuk-frontend/src/govuk/all.mjs#L88) giving us:

```mjs
/**
 * @typedef {import('govuk-frontend').Config} Config - Config for all components via `initAll()`
 * @typedef {import('govuk-frontend').ConfigKey} ConfigKey - Component config keys, e.g. `accordion` and `characterCount`
 */
```

As an example, you'll see from the [`ConfigKey`](https://github.com/alphagov/govuk-frontend/blob/main/shared/helpers/puppeteer.js#L257) tooltip the available configs:

<img width="938" alt="Tooltip for component config types" src="https://github.com/alphagov/govuk-frontend/assets/415517/d1e6eb1c-9d07-42be-9377-5d41d2f8f248">

<br>Which we can use here too:

```suggestion
   * @param {Config[ConfigKey]} config
```

Reply:
Hopefully we can get generics working with JSDoc and narrow those configs down by component